pycheribuild/projects/cross/juliet_test_suite.py

```diff
@@ -111,12 +111,46 @@ class BuildJulietCWE121(BuildJulietCWESubdir):
                                              subdirectory="testcases/CWE121_Stack_Based_Buffer_Overflow")
 
 
+class BuildJulietCWE122(BuildJulietCWESubdir):
+    target = "juliet-cwe-122"
+    cwe_number = 122
+    repository = ReuseOtherProjectRepository(BuildJulietTestSuite,
+                                             subdirectory="testcases/CWE122_Heap_Based_Buffer_Overflow")
+
+
+class BuildJulietCWE124(BuildJulietCWESubdir):
+    target = "juliet-cwe-124"
+    cwe_number = 124
+    repository = ReuseOtherProjectRepository(BuildJulietTestSuite,
+                                             subdirectory="testcases/CWE124_Buffer_Underwrite")
+
+
 class BuildJulietCWE126(BuildJulietCWESubdir):
     target = "juliet-cwe-126"
     cwe_number = 126
     repository = ReuseOtherProjectRepository(BuildJulietTestSuite, subdirectory="testcases/CWE126_Buffer_Overread")
 
 
+class BuildJulietCWE127(BuildJulietCWESubdir):
+    target = "juliet-cwe-127"
+    cwe_number = 127
+    repository = ReuseOtherProjectRepository(BuildJulietTestSuite, subdirectory="testcases/CWE127_Buffer_Underread")
+
+
+class BuildJulietCWE134(BuildJulietCWESubdir):
+    target = "juliet-cwe-134"
+    cwe_number = 134
+    repository = ReuseOtherProjectRepository(BuildJulietTestSuite, subdirectory="testcases/CWE134_Uncontrolled_Format_String")
+    cwe_warning_flags = ["-Wno-error=format-security"]
+
+
+class BuildJulietCWE188(BuildJulietCWESubdir):
+    target = "juliet-cwe-188"
+    cwe_number = 188
+    repository = ReuseOtherProjectRepository(BuildJulietTestSuite, subdirectory="testcases/CWE188_Reliance_on_Data_Memory_Layout")
+    cwe_warning_flags = ["-cheri-bounds=subobject-safe"]
```

Comment:
This is not really a warning flag and will only work when building for purecap, maybe just override setup to something like this:
```
def setup():
    super().setup()
    if self.compiling_for_cheri():
        self.CFLAGS.append("-cheri-bounds=subobject-safe")
```